app/src/main/java/in/co/rajkumaar/amritarepo/activities/LaunchingActivity.java

```diff
@@ -576,19 +529,133 @@ private void intentSemActivity(int position, String title) {
+                    }
+                }
+            }
+        });
     }
 
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == request_code && resultCode == ActivityResult.RESULT_IN_APP_UPDATE_FAILED) {
+                Utils.showToast(this,getString(R.string.update_failed));
+                appUpdateInfoTask.addOnSuccessListener(appUpdateInfo -> {
+                    if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE){
+                        if(appUpdateInfo.updatePriority() > 3){
+                            try {
+                                appUpdateManager.startUpdateFlowForResult(
+                                        // Pass the intent that is returned by 'getAppUpdateInfo()'.
+                                        appUpdateInfo,
+                                        // Or 'AppUpdateType.FLEXIBLE' for flexible updates.
+                                        AppUpdateType.IMMEDIATE,
+                                        // The current activity making the update request.
+                                        this,
+                                        // Include a request code to later monitor this update request.
+                                        request_code);
+                            } catch (IntentSender.SendIntentException e) {
+                                e.printStackTrace();
+                            }
+                        }
+                        else{
+                            try {
+                                appUpdateManager.startUpdateFlowForResult(
+                                        // Pass the intent that is returned by 'getAppUpdateInfo()'.
+                                        appUpdateInfo,
+                                        // Or 'AppUpdateType.FLEXIBLE' for flexible updates.
+                                        AppUpdateType.FLEXIBLE,
+                                        // The current activity making the update request.
+                                        this,
+                                        // Include a request code to later monitor this update request.
+                                        request_code);
+                            } catch (IntentSender.SendIntentException e) {
+                                e.printStackTrace();
+                            }
+                        }
+                    }
+                });
+        }
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        appUpdateManager
+                .getAppUpdateInfo()
+                .addOnSuccessListener(appUpdateInfo -> {
+                    // If the update is downloaded but not installed,
+                    // notify the user to complete the update.
+                    if(appUpdateInfo.updatePriority()>3){
```

Comment:
Please remove all priority dependent code, as I've no control over it from the DevConsole.

Reply:
There are two types of updates flexible and immediate. An Intermediate update is done if the user interaction or data is affected. If only the UI is updated we use Flexible update. I thought to change Flexible and Immediate based on priority in dev console(0-5). If greater than 3, I thought of doing Immediate otherwise Flexible.

Reply:
Yes, I understand. But the problem is this priority value is configurable only through their API and **not** from DevConsole. Hence, I suggested you to go with IMMEDIATE if the update is stalled for more than a day, otherwise keep it FLEXIBLE.

Ref : https://stackoverflow.com/a/61465476/8622362

Reply:
I will use stalledForMoreThanADay